src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorSourceUpdater.cs

```diff
@@ -48,59 +56,332 @@ public async Task<IUpdaterResult> IsApplicableAsync(IUpgradeContext context, Imm
                 throw new ArgumentNullException(nameof(context));
             }
 
-            await GetDiagnosticsAsync(context, inputs, token).ConfigureAwait(false);
+            var project = GetProjectWithGeneratedCode(context.CurrentProject.Required(), inputs);
+            _logger.LogDebug("Running upgrade analyzers on Razor files in {ProjectName}", project.Name);
+
+            // Use mapped locations so that we only get a number of diagnostics corresponding to the number of cshtml locations that need fixed.
+            var mappedDiagnostics = await GetDiagnosticsFromTargetFilesAsync(project, context, inputs.Select(GetGeneratedFilePath), new LocationAndIDComparer(true), token).ConfigureAwait(false);
+            _logger.LogInformation("Identified {DiagnosticCount} fixable diagnostics in Razor files in project {ProjectName}", mappedDiagnostics.Count(), project.Name);
+            var diagnosticsByFile = mappedDiagnostics.GroupBy(d => d.Location.GetMappedLineSpan().Path);
+            foreach (var diagnosticsGroup in diagnosticsByFile)
+            {
+                _logger.LogInformation("  {DiagnosticsCount} diagnostics need fixed in {FilePath}", diagnosticsGroup.Count(), diagnosticsGroup.Key);
+            }
+
+            return new FileUpdaterResult(mappedDiagnostics.Any(), diagnosticsByFile.Select(g => g.Key));
+        }
+
+        public async Task<IUpdaterResult> ApplyAsync(IUpgradeContext context, ImmutableArray<RazorCodeDocument> inputs, CancellationToken token)
```

Comment:
Can you split this method up into smaller methods for each step it's doing?

Reply:
Good idea.,